types/core/hooks.d.ts

```diff
@@ -117,72 +127,255 @@ declare class Hooks {
+     * @param contextOptions - the already created ContextMenuOptions
+     * @see {@link SceneNavigation#activateListeners}
+     */
+    getSceneNavigationContext: (jq: JQuery, contextOptions: ContextMenu.Option[]) => void;
+
+    /**
+     * This is called after getting the {@link ContextMenu} options for a {@link PlayerList} user, but before creating
+     * the ContextMenu.
+     * @param jq             - the JQuery of the ContextMenu parent element
+     * @param contextOptions - the already created ContextMenuOptions
+     * @remarks The name is 'getUserContextOptions'.
+     * @see {@link PlayerList#activateListeners}
+     */
+    getUserContextOptions: (jq: JQuery, contextOptions: ContextMenu.Option[]) => void;
+
+    /**
+     * This is called during the drop portion of a drag-and-drop event on the hotbar.
+     * @param hotbar - the Hotbar
+     * @param data   - the dropped data, already parsed as an object via JSON
+     * @param slot   - the slot of the macro target
+     * @returns whether additional callbacks should be called after this
+     * @see {@link Hotbar#_onDrop}
+     */
+    hotbarDrop: (hotbar: Hotbar, data: object, slot: string) => boolean;
+
+    /**
+     * This is called before the {@link Game} is initialized for the current window location.
+     * @see {@link Game#initialize}
+     */
+    init: () => void;
+
+    /**
+     * This is called when initializing shaders for a {@link PointSource}.
+     * @param pointSource   - the PointSource to initialize shaders for
+     * @param animationType - a key used in `CONFIG.Canvas.lightAnimations`
+     * @see {@link PointSource#_initializeShaders}
+     */
+    initializePointSourceShaders: (pointSource: PointSource, animationType: string) => void;
+
+    /**
+     * This is called after refreshing the {@link LightingLayer}.
+     * @param layer - the LightingLayer
+     * @see {@link LightingLayer#refresh}
+     */
+    lightingRefresh: (lighting: LightingLayer) => void;
+
+    /**
+     * This is called when the values of a {@link Token} are updated and before updating the values of the associated
+     * {@link Actor}.
+     * @param updateInfo - the raw update information
+     * @param attribute  - the attribute path
+     * @param isBar      - whether the new value is part of an attribute bar, or just a direct value
+     * @param isDelta    - whether the number represents a relative change (true) or an absolute change (false)
+     * @param value      - the target attribute value
+     * @param update     - the same object data, that will be passed to {@link Actor#update}
+     * @returns whether the Actor should be updated
+     * @see {@link Actor#modifyTokenAttribute}
+     * @see {@link Actor#update}
+     */
+    modifyTokenAttribute: (
+      updateInfo: {
+        attribute: string;
+        isBar: boolean;
+        isDelta: boolean;
+        value: number;
+      },
+      update: Record<string, number>
+    ) => boolean;
```

Comment:
Worth pulling into a separate type for later?

```suggestion
    modifyTokenAttribute: (
      updateInfo: ModifyTokenUpdateInfo,
      update: Record<string, number>
    ) => boolean;
```